utils/warnings.ml

```diff
@@ -349,10 +349,6 @@ let descriptions = [
       "Two labels or constructors of the same name are defined in two\n\
       \    mutually recursive types.";
     since = None };
-  { number = 31;
-    names = ["module-linked-twice"];
-    description = "A module is linked twice in the same executable.";
-    since = since 4 0 };
```

Comment:
I would like to understand well what is going on with the warning from a compatibility perspective:

- what happens if a user manually specifies this warning, on the command-line or in an attribute, by number or by name?
- what is shown in `-warn-help`?

My current understanding is as follows:

- Warning modifiers are interpreted by mutating dense flag arrays of all warning numbers.
- Using a number-based modifier `-31` or `+31` will modify the flag array and do nothing else (as the warning is never raised anymore). In practice this is silently ignored, which is probably fine.
- Using `-module-linked-twice` will result in a lookup failure because the `description` entry is removed here, which will eventually result in a confusing `ocaml_deprecated_cli` alert. This is not good.
- The warning will not appear in `-warn-help` anymore, because the description entry is removed.

I think that we should do this instead: keep the description entry, but extend the description to say that the warning is now ignored and replaced by a hard error.

My understanding is that we would then get the following behavior:
- specifying the warning by number or by name is ignored
- the warning shows up in `-warn-help`, with a mention that it is ignored

I think that this new behavior would be reasonable, a better default.

One may think of something even more informative (if users try to *disable* the warning, letting them know that this does not work anymore through... another warning?), but I think that it would not be worth the implementation and conceptual complexity.

Minor note: the definition of `defaults_warn_error` further down still mentions `+31`, it should be removed I think.

Reply:
I've changed it to
```
 31 [module-linked-twice] A module is linked twice in the same executable.
    Ignored: now a hard error (since 5.0).
```

Reply:
5.1 rather than 5.0 ?

Reply:
@Octachron, fixed thanks.